BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -163,6 +167,40 @@ - (void)viewDidAppear:(BOOL)animated
         [self handleConnectionErrorWithError:self.connectionError
                                      message:self.connectionErrorMessage];
     }
+
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(applicationWillResignActive:) name:UIApplicationWillResignActiveNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(applicationDidBecomeActive:) name:UIApplicationDidBecomeActiveNotification object:nil];
+}
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    if (self.backgroundTaskID != UIBackgroundTaskInvalid) {
+        [[UIApplication sharedApplication] endBackgroundTask:self.backgroundTaskID];
+        self.backgroundTaskID = UIBackgroundTaskInvalid;
+    }
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationWillResignActiveNotification object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationDidBecomeActiveNotification object:nil];
+
+    [super viewWillDisappear:animated];
+}
+
+// Fixes a bug starting iOS 11.3, where backgrounding the app to respond to
+// 2FA could cause the login to fail.
+- (void)applicationWillResignActive:(NSNotification *)note
+{
+    UIApplication *app = [UIApplication sharedApplication];
+    [app beginBackgroundTaskWithExpirationHandler:^{
+        [app endBackgroundTask:self.backgroundTaskID];
```

Comment:
weak self?

Reply:
Shouldn't need it. There should be no way to deallocate self before this block is released.